app/src/main/java/com/cornellappdev/android/eatery/ui/screens/UpcomingMenuScreen.kt

```diff
@@ -185,11 +186,13 @@ fun UpcomingMenuScreen(
                                 }
                             },
                             onFilterClicked = { filter ->
-                                if (viewState.selectedFilters.contains(filter)
-                                ) {
-                                    upcomingViewModel.removeLocationFilter(filter)
-                                } else {
-                                    upcomingViewModel.addLocationFilter(filter)
+                                (filter as? Filter.FromEatery)?.let {
```

Comment:
This confused me a bit at first but I take it that you just do this because it should be a `FromEatery` right? Since all the filters should be that?

Maybe leave a brief comment for that here :)

Also uh since this is def somewhat nontrivial logic can we make this call just some `onFilterClicked` from the viewmodel and move this logic there?

Reply:
Oh also--quick UX issue--I don't think users should be allowed to select BOTH north and west, for example. Only one location filter should be possible at a time. Since there's no eatery that is on two locations, right?

So let's also beef up the logic here to make it so only one of north, west, or central is active.

Reply:
I'd rather address that in a future PR, since most of the other screens have this same ability where you can select North, West, and Central at the same time